### PR TITLE
feat: add reduceCalc for usage without PostCSS

### LINKED ISSUE #298: [Feature Request]: Usage without postcss
### What should be improved?

Are you open to also making the library functionally similar to `@csstools/css-calc`? I tested both and the output for `postcss-calc` is better.

### Describe the solution you would like

Something like
```js
import { calc } from 'postcss-calc'

console.log(calc('calc(10 * 2)')) // 20
```

### Possible alternatives

_No response_

### Additional context

_No response_

### Are you willing to work on this?

- [ ] Yes, I would like to help

### PR DESCRIPTION
**Pull request by phuoccss**
Fixes https://github.com/postcss/postcss-calc/issues/298

### CODE REVIEW

**Review by ludofischer — CHANGES_REQUESTED** [review]
Could you add a test that exercises the API you would like to use? So that we don't change it inadvertently in the future.

**Comment by phuoccss** [comment]
Added. I mostly mirrored cases from plugin unit test.

File: test/unit/reduceCalc.test.mjs [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+      warnWhenCannotResolve: true,
+    });
+    assert.equal(output, 'red');
+    assert.equal(warnings.length, 0);
+    assert.equal(reduceCalc('10px 20px'), '10px 20px');
+  });
+});
+
+// --- Bare math functions (issue #189) -----------------------------------
+describe('reduceCalc: bare math functions', () => {
```
Comment (ludofischer) [inline test/unit/reduceCalc.test.mjs:261]: Aren't tests starting from here a to the end of the file a bit redundant? It understand testing option combinations as they might differ from the postcss plugin and they were touched by the change, but these tests below seem almost duplicates of existing ones.

**Review by ludofischer — CHANGES_REQUESTED** [review]
Could you check you did not add too many redundant tests?

File: test/unit/reduceCalc.test.mjs (hunk 1 above)
Comment (phuoccss) [reply test/unit/reduceCalc.test.mjs:261]: Why do you think they are redundant? Those seems to be non-PostCSS edge cases that the new, lower-level function have to cover.

What do you think if we remove those tests in the plugin suite instead? That means the plugin suite is to ensure that the source css is processed correctly given different plugin option combinations. There's no need to cover all edge cases etc.

Comment (ludofischer) [reply test/unit/reduceCalc.test.mjs:261]: Maybe it's better to understand why tests are failing first.

Comment (phuoccss) [reply test/unit/reduceCalc.test.mjs:261]: Was a typo.

**Review by ludofischer — APPROVED** [review]
